dot/sync/chain_processor.go

```diff
@@ -83,17 +83,19 @@ func (s *chainProcessor) processReadyBlocks() {
 		}
 
 		if err := s.processBlockData(bd); err != nil {
-			logger.Errorf("block data processing for block with hash %s failed: %s", bd.Hash, err)
-
 			// depending on the error, we might want to save this block for later
 			if errors.Is(err, errFailedToGetParent) {
+				logger.Tracef("block data processing for block with hash %s failed: %s", bd.Hash, err)
 				if err := s.pendingBlocks.addBlock(&types.Block{
 					Header: *bd.Header,
 					Body:   *bd.Body,
 				}); err != nil {
 					logger.Debugf("failed to re-add block to pending blocks: %s", err)
 				}
+				continue
 			}
+
+			logger.Errorf("block data processing for block with hash %s failed: %s", bd.Hash, err)
```

Comment:
nit Maybe swap around the condition for less indentation?

```suggestion
			if !errors.Is(err, errFailedToGetParent) {
				logger.Errorf("block data processing for block with hash %s failed: %s", bd.Hash, err)
				continue
			}

			logger.Tracef("block data processing for block with hash %s failed: %s", bd.Hash, err)
			if err := s.pendingBlocks.addBlock(&types.Block{
				Header: *bd.Header,
				Body:   *bd.Body,
			}); err != nil {
					logger.Debugf("failed to re-add block to pending blocks: %s", err)
				}
			}
```